scripts/bluetooth/mesh/mesh_dfu_metadata.py

```diff
@@ -0,0 +1,425 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+
+'''
+This script is used to extract the Bluetooth Mesh composition data from the build folder of
+an application. This information is used to generate a hash that can verify if the composition
+of a device will change after a Mesh device firmware upgrade has completed.
+The input to this script is the following content of the build folder of an application:
+    - build/zephyr/zephyr.elf
+    - build/zephyr/.config
+The script produces the following output to the build folder:
+	- build/zephyr/dfu_application.zip_mesh_metadata.json
+This output is also appended to the archive located at build/zephyr/dfu_application.zip.
+'''
+
+import struct
+import sys
+import os
+from elftools.elf.elffile import ELFFile
+import json
+from cryptography.hazmat.primitives import cmac
+from cryptography.hazmat.primitives.ciphers import algorithms
+from zipfile import ZipFile
+
+
+class Elem:
+    def __init__(self, loc):
+        self.loc = loc
+        self.vnd_list = []
+        self.sig_list = []
+
+    def vnd_model_add(self, cid, vid):
+        self.vnd_list.append((cid << 16) + vid)
+
+    def sig_model_add(self, id):
+        self.sig_list.append(id)
+
+    def bytestring_generate(self):
+        bytestring = bytearray()
+        bytestring.extend(self.loc.to_bytes(2, 'little'))
+        bytestring.append(len(self.sig_list))
+        bytestring.append(len(self.vnd_list))
+
+        for sig in self.sig_list:
+            bytestring.extend(sig.to_bytes(2, 'little'))
+        for vnd in self.vnd_list:
+            bytestring.extend(vnd.to_bytes(4, 'little'))
```

Comment:
Should we check system endianness or little endianness is always the case?

Reply:
It is always little I think.

Reply:
Seems like it is possible to get this info: https://github.com/eliben/pyelftools/blob/main/elftools/elf/elffile.py#L65

Reply:
Don't bother I thought this part is about reading the raw data.